EVM and High Level Languages.md

```diff
@@ -0,0 +1,95 @@
+# The EVM and High-Level Smart Contract Programming Languages
+
+## Introduction
```

Comment:
I'm not sure how much we need the introduction, this post is an EVM deep-dive, so I think people know what it is.

Reply:
I gutted this and re-wrote 75% of it, please review.  Thanks!